pycommon/report2idea.py

```diff
@@ -78,6 +107,59 @@ def signal_h(signal, f):
     global trap
     trap.terminate()
 
+def check_valid_timestamps(idea, dpast, dfuture):
+    """
+    Return True if EventTime, CeaseTime, and DetectTime are in the interval (CreateTime - dpast, CreateTime + dfuture).
+
+    :param idea: dict with filled IDEA message
+    :param dpast: int maximal number of days into past
+    :param dfuture: int maximal number of days into future
+
+    >>> idea = {"CreateTime": "2019-03-11T15:00:00Z", "DetectTime": "2019-03-11T14:59:00Z",
+    ... "EventTime": "2019-03-11T14:57:00Z", "CeaseTime": "2019-03-11T14:58:50Z"}
+    >>> check_valid_timestamps(idea, 1, 0)
+    True
+    >>> idea = {"CreateTime": "2019-03-11T15:00:00Z", "DetectTime": "2019-03-11T15:00:00Z",
+    ... "EventTime": "2019-03-11T14:57:00Z", "CeaseTime": "2019-03-11T14:58:50Z"}
+    >>> check_valid_timestamps(idea, 1, 0)
+    True
+    >>> # One day tolerance for the future
+    >>> idea = {"CreateTime": "2019-03-11T15:00:00Z", "DetectTime": "2019-03-11T15:10:00Z",
+    ... "EventTime": "2019-03-11T14:57:00Z", "CeaseTime": "2019-03-11T14:58:50Z"}
+    >>> check_valid_timestamps(idea, 1, 1)
+    True
+    >>> # DetecTime from future
+    >>> idea = {"CreateTime": "2019-03-11T15:00:00Z", "DetectTime": "2019-03-11T15:10:00Z",
+    ... "EventTime": "2019-03-11T14:57:00Z", "CeaseTime": "2019-03-11T14:58:50Z"}
+    >>> check_valid_timestamps(idea, 1, 0)
+    False
+    >>> # CreateTime from past
+    >>> idea = {"CreateTime": "2019-03-10T15:00:00Z", "DetectTime": "2019-03-11T14:59:00Z",
+    ... "EventTime": "2019-03-11T14:57:00Z", "CeaseTime": "2019-03-11T14:58:50Z"}
+    >>> check_valid_timestamps(idea, 1, 0)
+    False
+
+    """
+    et = idea.get("EventTime", None)
+    ct = idea.get("CeaseTime", None)
+    dt = idea.get("DetectTime", None)
+    create = idea.get("CreateTime", None)
+    if not create:
+        return True
+    else:
+	create = parseRFCtime(create)
+    deltapast = create - timedelta(dpast)
+    deltafuture = create + timedelta(dfuture)
+    for t in [et, ct, dt]:
+        if t:
+            # convert timestamp to datetime
+	    t = parseRFCtime(t)
+	    # check allowed interval
```

Comment:
Lines 150, 156 and 157 contain tabs instead of spaces.

Reply:
fixed in 6c31616